x-pack/plugin/monitoring/src/test/java/org/elasticsearch/xpack/monitoring/collector/shards/ShardsCollectorTests.java

```diff
@@ -130,15 +144,60 @@ public void testDoCollect() throws Exception {
     private static RoutingTable mockRoutingTable() {
```

Comment:
The test setup here is starting to feel a bit messy, not sure what can be done about it.